api/tests/unit/mappers/test_unit_mappers_dynamodb.py

```diff
@@ -0,0 +1,203 @@
+import uuid
+from datetime import datetime
+from decimal import Decimal
+from typing import TYPE_CHECKING, Any, List, Optional
+
+import pytest
+from flag_engine.features.models import (
+    MultivariateFeatureOptionModel,
+    MultivariateFeatureStateValueList,
+    MultivariateFeatureStateValueModel,
+)
+from pydantic import BaseModel, Field
+from pytest_mock import MockerFixture
+
+from mappers import dynamodb
+
+if TYPE_CHECKING:  # pragma: no cover
+    from environments.identities.models import Identity
+    from environments.models import Environment, EnvironmentAPIKey
+
+
+@pytest.fixture
+def mock_engine_model() -> BaseModel:
+    class TestModel(BaseModel):
+        field: str
+
+    return TestModel(field="foobar")
+
+
+def test_map_environment_to_environment_document__call_expected(
+    environment: "Environment",
+    mocker: MockerFixture,
+    mock_engine_model: BaseModel,
+) -> None:
+    # Given
+    map_environment_to_engine_mock = mocker.patch(
```

Comment:
I'm not too clear on why we need to rely on mocking so heavily here?

Reply:
This has been fixed.